snapshot-labs/subgrapher
# feat: handle delegation subgraphs and hide subgraph API key in env

Primary goal is to hide Subgraph API key that is public here https://github.com/snapshot-labs/snapshot.js/blob/master/src/delegationSubgraphs.json#L2

### Summary
- Move delegation subgraphs from https://github.com/snapshot-labs/snapshot.js/blob/master/src/delegationSubgraphs.json
- Hide API key in env

### How to test
- Add `SUBGRAPH_KEY=0f15b42bdeff7a063a4e1757d7e2f99e` to .env
- If we send a request to a new URL format, for example `'http://localhost:3000/delegation/1` or `'http://localhost:3000/delegation/10` should work,  try the below curl:
```bash
curl --location 'http://localhost:3000/delegation/1' \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{"query":"query { delegations (where: {delegate: \"0x24f15402c6bb870554489b2fd2049a85d75b982f\"}, first: 1000) { delegator space } }","variables":{}}'
```
- Everything should work the same as before, for example `http://localhost:3000/api.thegraph.com/subgraphs/name/snapshot-labs/snapshot-goerli` , try below curl
```bash
curl --location 'http://localhost:3000/api.thegraph.com/subgraphs/name/snapshot-labs/snapshot-optimism' \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{"query":"query { delegations (where: {delegate: \"0x24f15402c6bb870554489b2fd2049a85d75b982f\"}, first: 1000) { delegator space } }","variables":{}}'
```

### To do:
- [x] Add `SUBGRAPH_KEY` to DO
